65 download cycle

broija opened the pull request:
closes #65

broija committed:
[FEAT][WIP] Storage limit (#17_enforce_storage_limit)
- Storage limit saved.
- Calculating remaining free space after each download.
- Added FreeSpaceExhausted component.

broija committed:
[FEAT] Storage limit (#17_enforce_storage_limit)
- Added user config.
- Added IPC message to check free space.
- Storage selection : reduce custom selection button sizes.
- FreeSpaceExhausted component :
  - Added buttons.
  - Turned into success component.
- DispatchRequestPayload : node_id is now a number.
- Dispatcher : free space sent is now remaining allocated space.

broija committed:
[FEAT] (#65_download_cycle)
- Now files are downloaded until allocated space limit is reached or nothing has been downloaded.
- node_id : temporarily switched back to string for API compatibility.
- Failed donwloads are now distinguished from others, but no failure handling yet.

kacime commented on src/renderer/hooks/useDownloadManager.ts:
```diff
@@ -104,10 +106,14 @@ export function useDownloadManager() : DownloadManager {
       if (!remainingFreeSpace) {
         callbacks.onError?.('Free space exhausted', undefined);
-        return;
+        return true; // broija 2025/09/12 : allocated space exhaustion is not an error
```
on va pas boucler à l'infini ici du coup au niveau du while ?

broija committed:
[FIX] (#17_enforce_storage_limit)
- UserConfig : create ~/.shelter-config.json when necessary.
- Simplified registerSystemInfo for bandwith limitation.

broija committed:
Merge remote-tracking branch 'origin/17_enforce_storage_limit' into 65_download_cycle

broija committed:
[FIX] (#65_download_cycle)
Fixed free space check returns in nextDownload and downloadAssets.
